docs/topics/strings.md

```diff
@@ -132,4 +141,48 @@ use the following syntax:
 val price = """
 ${'$'}_9.99
 """
-```
\ No newline at end of file
+```
+
+## String formatting
+
+> String formatting with the `String.format()` function is available for the Kotlin/JVM platform.
+>
+{type="note"}
+
+To format a string to your specific requirements, use the [`String.format()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/format.html) function. 
+
+The `String.format()` function accepts a format string and one or more arguments. The format string contains one placeholder 
+(`%`) for each remaining argument, followed by format specifiers. Format specifiers are formatting instructions for the 
+respective argument. In the output, each argument fills its corresponding placeholder in the defined format:
+
+```kotlin
+fun main() { 
+//sampleStart
+    // Formats to add zeroes and make a length of seven
+    val integerNumber = String.format("%07d", 31416)
+    println(integerNumber)
+    // 0031416
+
+    // Formats with four decimals and sign
+    val floatNumber = String.format("%+.4f", 3.141592)
+    println(floatNumber)
+    // +3.1416
+
+    // Formats with uppercase for two placeholders
+    val helloString = String.format("%S %S", "hello", "world")
+    println(helloString)
+    // HELLO WORLD
+//sampleEnd    
+}
+```
+{interpolate-variables="false" kotlin-runnable="true" kotlin-min-compiler-version="1.3"}
+
+The `String.format()` function provides similar functionality to string templates. In comparison, the 
+`String.format()` function is more versatile because you have many options for formatting. 
+
+Besides, the `String.format()` function operates during runtime. Since the format string is not constant, it can come from 
+a variable. This characteristic is useful when the format string changes, for example, in localization cases that depend 
+on the user locale.
+
+In contrast, the `String.format()` function is also more error-prone than string templates because it is easy to 
+mismatch the number or position of the arguments with their corresponding placeholders.
```

Comment:
```suggestion
Be careful when using the `String.format()` function because the additional flexibility can make your code more prone to errors. It can be easy to 
mismatch the number or position of the arguments with their corresponding placeholders.
```

Reply:
I removed "additional flexibility" from here according to Ilya's previous explanation.